### ISSUE TITLE
mintty crashes after selecting terminal tab in options window

### ISSUE THREAD

**Issue by Biswa96** [mintty/mintty#1185]
Open mintty > Right click > Options > Terminal crashes the running program like following

![image](https://user-images.githubusercontent.com/31443074/202907292-dba0f9b2-2920-4294-8e15-f96024d418e7.png)

**Comment by mintty** [comment]
I had this for a while during development but not in a release version, including current 3.6.2. Did you check out something from the git history?

**Comment by Biswa96** [comment]
I am using mintty 3.6.2 (x86_64-pc-msys) in msys2. What should I check?

**Comment by Biswa96** [comment]
I have compiled mintty from git repository and placed it in msys64/usr/bin directory and got same error.

**Comment by mintty** [comment]
I do not reproduce this on msys2. Any special configuration?

**Comment by Biswa96** [comment]
<details><summary>Click here to see my minttyrc file</summary>

```ini
ForegroundColour=255,255,255
ThemeFile=helmholtz
FontHeight=11
Font=Consolas
FontWeight=400
FontIsBold=no
FontSmoothing=full
BoldAsFont=yes
BoldAsColour=yes
ConfirmExit=no
Term=xterm-256color
CharNarrowing=200
AllowBlinking=yes
Columns=120
Rows=27
ProgressBar=yes
CtrlShiftShortcuts=yes
RewrapOnResize=yes
```

</details>

**Comment by mintty** [comment]
Please try the attached debug version and upload the stacktrace.
[mintty-msys-debug.zip](https://github.com/mintty/mintty/files/10050249/mintty-msys-debug.zip)

**Comment by Biswa96** [comment]
Now here is the twist. I can not reproduce the issue with that debug build. Also I get same crash in cygwin's mintty.

**Comment by BrianInglis** [comment]
> I get same crash in cygwin's mintty.

I have no such problems with 3.6.2 x86/_64 either with right-click, menu key, or window icon right-click, and Terminal, or any other selection. 
Is there anything that might require or be faulted by cygwin-console-helper?

**Comment by Biswa96** [comment]
Feel free to close the issue if no one can reproduce it. The issue maybe in my system or cygwin/msys2 setup.

**Comment by mintty** [comment]
Do you have the issue with older releases?

**Comment by Biswa96** [comment]
Yes, older binaries also crashes with same window as in OP. As I said before, the issue maybe in my system or cygwin/msys2 setup.

**Comment by mintty** [comment]
Also please run `MINTTY_DEBUG=o mintty` and check for stdout and additional message boxes.

**Comment by Biswa96** [comment]
<details><summary>Click there to see the output of that command</summary>

```
$ MINTTY_DEBUG=o mintty
WM_NOTIFY
WM_NOTIFY: GetSelection
WM_NOTIFY: Destroy
WM_NOTIFY: cleanup
WM_NOTIFY: create
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: refresh
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_USER
WM_USER: lookup
WM_USER: handler
WM_USER: end
WM_NOTIFY
WM_NOTIFY: GetSelection
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: Destroy
WM_NOTIFY: cleanup
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_NOTIFY: create
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
catch 11: WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
WM_COMMAND
WM_COMMAND: handle
WM_COMMAND: end
Segmentation fault (core dumped)
```

</details>

**Comment by mintty** [comment]
Symptoms are duplicate of #610. Apparently there was trouble with configuration file on network drive, do you have that?

**Comment by Biswa96** [comment]
I have no network drive. I have provided the .minttyrc file in previous comment.

**Comment by mintty** [comment]
Seems to happen during dlg_refresh, called from windialog.c. I'd try printf debugging in winctrls.c but that won't help if your own build does not expose the issue.

**Comment by Biswa96** [comment]
🤷‍♂️

**Comment by dscho** [comment]
@Biswa96 https://github.com/git-for-windows/git/issues/4065#issuecomment-1333274607 mentioned something about a disabled Printer Spooler causing this. Maybe the same is true for you?

**Comment by mintty** [comment]
Bug reproduced (after stopping Print Spooler service), thanks for the hint.

**Comment by mintty** [comment]
Interpreting the description of Windows function EnumPrinters strictly, it should return byte count and item count of 0 if it fails but it doesn't, returns undefined instead, which causes wrong subsequent handling.
Reoccurrence of #149 and #168.

**Comment by Biswa96** [comment]
Thanks for fixing the issue. I have compiled mintty from latest git commit and it fixes the crash in terminal tab.

Windows API is awesome, isn't it? 😄

**Comment by mintty** [comment]
Released 3.6.3.
